docs/swift/DesignDoc.md

```diff
@@ -224,6 +245,16 @@ drafted or sent proposals for review that would benefit our designs:
 *   [Drafted: Resettable Properties]
     (https://github.com/patters/swift-evolution/blob/master/proposals/0000-resettable-properties.md)
 
+Lastly, we note some name transformations that we would apply to fields:
+
+* Repeated fields will have `Array` appended to their names. This is because
+  there is a trend in some messages to name repeated fields with the singular
+  version of the name (`descriptor.proto` in particular does this), and
+  `foo.name.append(bar)` looks potentially like appending a string to another
+  string whereas `foo.nameArray.append(bar)` is much clearer.
+* If a field name would collide with one of the properties or methods in
+  `ProtoMessage`'s public interface, we append `Field` to its name.
```

Comment:
Simply including an underscore here would certainly fix the common case.  There are some corner cases that would need to be documented, though:

     optional int32 _field__1 = 7;

Leading underscores and multiple underscores within field names are handled inconsistently elsewhere in the protobuf codebase; documenting this would certainly be a service to the community.